src/common/slice.rs

```diff
@@ -43,23 +47,24 @@ impl<'a, T> From<&'a [T]> for UncheckedSlice<T> {
 }
 
 #[derive(Debug)]
+#[repr(C)]
 pub struct BoundedSlice<T> {
-    data: UncheckedSlice<T>,
-    len: usize,
+    pub data: UncheckedSlice<T>,
```

Comment:
Does it really need to be public other than for debugging proposes?

Reply:
If so, it might be better to make it private and comment the public access on the `print_instance_offsets`.
Thoughts?

Reply:
It's just for debugging purpose only.

Reply:
Maybe we don't really need print_instance_offsets.